libs/execution_context/ebpf_maps.c

```diff
@@ -43,33 +43,124 @@ typedef struct _ebpf_core_object_map
 // Define the granularity at which the LRU list is updated.
 #define EBPF_LRU_GENERATION_COUNT 10
 
+// Limit maximum number of partitions to 8 as a trade-off between memory usage and performance.
+// Fewer partitions will result in more contention on the lock, but more partitions will consume more memory.
+#define EBPF_LRU_MAXIMUM_PARTITIONS 8
+
 /**
- * @brief Each LRU entry tracks a key for an entry stored in a LRU map. The entry is stored in one of two lists: hot or
- * cold. The hot list is for items that have been accessed in the current generation, where a generation is defined as a
- * period of time during which max_entries/EBPF_LRU_GENERATION_COUNT elements have been accessed in the map. The cold
- * list is for items that have not been accessed in the current generation. When the hot list reaches
+ * @brief The BPF_MAP_TYPE_LRU_HASH is a hash table that stores a limited number of entries. When the map is full, the
+ * least recently used entry is removed to make room for a new entry. The map is implemented as a hash table with a pair
+ * of double-linked lists per partition. The hot list contains entries that have been accessed in the current
+ * generation, and the cold list contains entries that have not been accessed in the current generation. When entries in
+ * the cold list are accessed, they are moved to the hot list. When the hot list reaches
  * max_entries/EBPF_LRU_GENERATION_COUNT, the hot list is merged into the cold list, a new generation is started, and
- * the hot list is cleared.  When space is needed in the map, the cold list is trimmed to make room for new entries. The
- * cold list is trimmed by removing the oldest entries in the list, which are part of the oldest generation. The benefit
- * is that the lock only needs to be acquired when moving items between generations, and not on every access.
+ * the hot list is cleared. When space is needed an entry is selected from a cold-list and is removed from the
+ * hash-table.
```

Comment:
```suggestion
 * hash table.
```
Consistency with line 51 etc.